packages/ui5-cc-spreadsheetimporter/src/control/SpreadsheetDialog.ts

```diff
@@ -39,6 +48,63 @@ export default class SpreadsheetDialog extends Dialog {
 		}
 	};
 
+	onAfterRendering(event: any) {
+		super.onAfterRendering(event);
+		const domRef = this.getDomRef();
+		domRef.addEventListener("dragover", this.handleDragOver.bind(this), false);
+		domRef.addEventListener("dragenter", this.handleDragEnter.bind(this), false);
+		domRef.addEventListener("dragleave", this.handleDragLeave.bind(this), false);
+		domRef.addEventListener("drop", this.handleFileDrop.bind(this), false);
```

Comment:
Hi Marian, I haven't checked it out completely but just wanted to drop this: I hope this doesn't lead to duplicate registrations in case the rendering is executed more often than than the resource is actually being destroyed again (remove listeners in exit event). You know the lifecycle of the control way better than I currently do so I just wanted to comment it. 😬

Reply:
In what cases would that be possible?
I forgot something so I need to fix someone in the anyway

Reply:
Hey @marianfoo, seems that I've been concerned for no reason. In your case the `exit` seems to not even be run, at least when the user closes the dialog regularly using the close button or <kbd>ESC</kbd>. Instead the entire DOM element is thrown away in UI5s Dialog itself [here](https://github.com/SAP/openui5/blob/f0902f9d2bc3545f083939abdcf7d4077c34b2d9/src/sap.m/src/sap/m/Dialog.js#L877-L886). I'm just always worried to have things registered multiple times in case the control does a re-render due to property or visual changes and the like, so I wanted to mentioned it briefly. Sorry for wasting your time! :)